contrib/src/main/java/com/datatorrent/contrib/enrich/FSLoader.java

```diff
@@ -0,0 +1,181 @@
+ * {"productCategory": 5, "productId": 2}
+ * {"productCategory": 5, "productId": 3}
+ * </p>
+ * Each line in a file should be a valid json object which represent the record and each key/value pair in that json object
+ * represents the fields/value.
+ * <p>
+ * NOTE: This loader should be used with caution as all the data present in the file is loaded in memory.
+ */
+@InterfaceStability.Evolving
+public class FSLoader extends ReadOnlyBackup
+{
+  @NotNull
+  private String fileName;
+
+  private transient Path filePath;
+  private transient FileSystem fs;
+  private transient boolean connected;
+
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final ObjectReader reader = mapper.reader(new TypeReference<Map<String, Object>>(){});
+  private static final Logger logger = LoggerFactory.getLogger(FSLoader.class);
+
+  public String getFileName()
+  {
+    return fileName;
+  }
+
+  public void setFileName(String fileName)
+  {
+    this.fileName = fileName;
+  }
+
+  @Override
+  public Map<Object, Object> loadInitialData()
+  {
+    Map<Object, Object> result = null;
+    FSDataInputStream in = null;
+    BufferedReader bin = null;
+    try {
+      result = Maps.newHashMap();
+      in = fs.open(filePath);
+      bin = new BufferedReader(new InputStreamReader(in));
+      String line;
+      while ((line = bin.readLine()) != null) {
+        try {
+          Map<String, Object> tuple = reader.readValue(line);
+          result.put(getKey(tuple), getValue(tuple));
+        } catch (JsonProcessingException parseExp) {
+          logger.info("Unable to parse line {}", line);
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (bin != null) {
+        IOUtils.closeQuietly(bin);
+      }
+      if (in != null) {
+        IOUtils.closeQuietly(in);
+      }
+      try {
+        fs.close();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    logger.debug("loading initial data {}", result.size());
+    return result;
+  }
+
+  private Object getValue(Map<String, Object> tuple)
+  {
+    ArrayList<Object> includeTuple = new ArrayList<Object>();
+    for (FieldInfo s : includeFieldInfo) {
+      includeTuple.add(tuple.get(s.getColumnName()));
+    }
+    return includeTuple;
+  }
+
+  private Object getKey(Map<String, Object> tuple)
+  {
+    ArrayList<Object> lst = new ArrayList<Object>();
+    for (FieldInfo key : lookupFieldInfo) {
+      lst.add(tuple.get(key.getColumnName()));
+    }
+    return lst;
+  }
+
+  @Override
+  public Object get(Object key)
+  {
+    return null;
+  }
+
+  @Override
+  public List<Object> getAll(List<Object> keys)
+  {
+    return null;
+  }
+
+  @Override
+  public void connect() throws IOException
+  {
+    Configuration conf = new Configuration();
+    this.filePath = new Path(fileName);
+    this.fs = FileSystem.newInstance(filePath.toUri(), conf);
```

Comment:
Considering the size of the file, I think it might be just more efficient to reload the file.
This FSLoader is to be used when size of the file is small even to be loaded in memory.